Cpp_test/lib/ipc-shm-receive.cpp

```diff
@@ -0,0 +1,110 @@
+#include "ipc-shm-receive.h"
+
+IPCShmReceive::~IPCShmReceive()
+{
+
+	shm_unlink(this->info.method_name);
+	if (pthread_cond_destroy(&(this->shm_ptr->cond_re)) != 0)
+	{
+		std::cerr << "Error at pthread_cond_destroy(): receive" << std::endl;
+	}
+	if (pthread_cond_destroy(&(this->shm_ptr->cond)) != 0)
+	{
+		std::cerr << "Error at pthread_cond_destroy(): send" << std::endl;
+	}
+	if (pthread_mutex_destroy(&(this->shm_ptr->mutex)) != 0)
```

Comment:
`It shall be safe to destroy an initialized mutex that is unlocked. Attempting to destroy a locked mutex results in undefined behavior.` https://linux.die.net/man/3/pthread_mutex_destroy

You should be sure to unlock it before destroying it.